src/main/java/view/ListView.java

```diff
@@ -128,6 +157,31 @@ private void buildActionListeners() {
                     }
                 }
         );
+
+        filterButton.addActionListener(
+                evt -> {
+                    if (evt.getSource().equals(filterButton)) {
+                        filterController.execute(
+                                filterViewModel.getState().getFilterCriteria(),
+                                listViewModel.getState().getCurrentCollectionType(),
+                                listViewModel.getState().getCurrentCollectionName()
+                        );
+                    }
+                }
+        );
+
+        clearButton.addActionListener(
+                evt -> {
+                    if (evt.getSource().equals(clearButton)) {
+                        filterPanelManager.clearFilterPanel();
+                        filterController.execute(
```

Comment:
Yeah, that makes sense. Good suggestion!

Reply:
Ah, that does make sense, since your filter use case updates the state information, and this would make the flow of control more consistent with the rest of the program.

On your second point, you don't need the unfiltered list of media names to display. What I would've done in your shoes is make your `FilterState`'s `filteredMediaNames` not final, and make it so if the variable is `null`, it means nothing should be filtered. And you can add `@Nullable` tag to the getter & update the getter/setter method's javadocs accordingly to document the behavior.

Reply:
That's a cool way of doing it, thanks. It hadn't occurred to me that the RowFilter could take in null.